lib/zlib.js

```diff
@@ -156,6 +156,28 @@ function flushCallback(level, strategy, callback) {
   }
 }
 
+function validateFiniteNumber(number, name) {
+  if (number === undefined || number !== number) {
+    return false;  // undefined or NaN
+  }
+
+  if (typeof number !== 'number') {
+    const err = new errors.TypeError('ERR_INVALID_ARG_TYPE', name,
+                                     'number', number);
+    Error.captureStackTrace(err, validateFiniteNumber);
+    throw err;
+  }
+
+  if (!Number.isFinite(number)) {
```

Comment:
Using `Number.isFinite(number)` will implicitly also test for `NaN` and for the type `number`. Therefore those two checks could be moved inside of this block to make the average case faster.

Reply:
Not sure I am following...NaN returns false here (no errors) while non-numbers throws a different error, how could we move those checks into this block?

Reply:
Oh, sorry, you can only move the `typeof number !== `number` in here. I overlooked that `NaN` should be accepted and gets a default value (do we really want that? I suggest to throw an error in that case as well).

The reason why the typecheck can be moved in here is:

`Number.isFinite(nonNumber) === false`. So the average case does not have to test for that. Only in case the `isFinite` test fails, it should be checked if it is the wrong type.

Reply:
I also doubt if it's necessary to use default values for NaNs, but several tests fail if I change that so I would like to investigate this behavior later.

I get what you were suggesting now. That makes sense to me, I'll change that later, thanks